lib/compiler-singlepass/src/machine.rs

```diff
@@ -305,17 +305,30 @@ impl Machine {
         }
     }
 
-    pub(crate) fn get_local_location(&self, idx: usize) -> Location {
+    pub(crate) fn get_local_location(&self, idx: u32) -> Location {
         // Use callee-saved registers for the first locals.
+        //
+        // NB: This calculation cannot reasonably overflow. `self.locals_offset` will typically be
+        // small (< 32), and `idx` is bounded to `51000` due to limits imposed by the wasmparser
```

Comment:
I think the way code is set up makes it quite apparent `self.locals_offset` will be very reasonably sized and largely constant, so the value adding an assert around it would have a negligible value, I feel?